provider/kubernetes/kubernetes.go

```diff
@@ -367,23 +372,35 @@ func (p *Provider) loadConfig(templateObjects types.Configuration) *types.Config
 	return configuration
 }
 
-func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) string {
+func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) (string, error) {
 	if len(pa.Path) == 0 {
-		return ""
+		return "", nil
 	}
 
 	ruleType := i.Annotations[label.TraefikFrontendRuleType]
 	if ruleType == "" {
 		ruleType = ruleTypePathPrefix
 	}
 
+	pathReplaced := false
+	if ruleType == ruleTypeReplacePath {
+		pathReplaced = true
+	}
+
 	rules := []string{ruleType + ":" + pa.Path}
 
 	if rewriteTarget := i.Annotations[annotationKubernetesRewriteTarget]; rewriteTarget != "" {
 		rules = append(rules, ruleTypeReplacePath+":"+rewriteTarget)
+		pathReplaced = true
```

Comment:
Shouldn't we return an error here already if `pathReplaced` is true at this point?